pkg/cloudprovider/vm.go

```diff
@@ -107,3 +108,10 @@ func (i *vmIdentityInfo) AppendUserIdentity(id string) bool {
 	}
 	return appendUserIdentity(&i.info.Type, i.info.IdentityIds, id)
 }
+
+func (i *vmIdentityInfo) CheckUserIdentityExists(id string) error {
+	if i.info.IdentityIds == nil || !checkIfIDInList(i.info.IdentityIds, id) {
+		return fmt.Errorf("Identity %s not found", id)
```

Comment:
nit: go style typically has errors start with lower case

Also, can this interface return a bool instead?